mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -497,6 +498,9 @@ private[ml] trait GBTParams extends TreeEnsembleParams with HasMaxIter with HasS
   @deprecated("This method is deprecated and will be removed in 3.0.0.", "2.1.0")
   def setStepSize(value: Double): this.type = set(stepSize, value)
 
+  /** @group setParam */
+  def setValidationIndicatorCol(value: String): this.type = set(validationIndicatorCol, value)
```

Comment:
Since we need this setter to be in the concrete classes for Java compatibility, don't bother putting it here.